assets/js/state/sagas/hosts.js

```diff
@@ -1,7 +1,38 @@
-import { put, takeEvery } from 'redux-saga/effects';
-import { HOST_DEREGISTERED, removeHost } from '@state/hosts';
+import { delay, put, race, call, take, takeEvery } from 'redux-saga/effects';
+import {
+  CHECK_HOST_IS_DEREGISTERABLE,
+  CANCEL_CHECK_HOST_IS_DEREGISTERABLE,
+  HOST_DEREGISTERED,
+  removeHost,
+  setHostListDeregisterable,
+} from '@state/hosts';
 import { notify } from '@state/actions/notifications';
 
+export function* markDeregisterableHosts(hosts) {
+  yield put(
+    setHostListDeregisterable(
+      hosts.filter(({ heartbeat }) => heartbeat !== 'passing')
+    )
+  );
+}
+
+function* hostDeregisterable({ debounce, host }) {
```

Comment:
I have excluded testing for `hostDeregisterable()` as it is the same as
the test for `checkHostDeregisterable()`, which simply calls it with a
cancellable condition. Thoughts on this?